app/providers/MyUsernamePasswordAuthProvider.java

```diff
@@ -410,25 +411,39 @@ protected String getPasswordResetMailingSubject(final User user,
 	protected Body getPasswordResetMailingBody(final String token,
 			final User user, final Context ctx) {
 
-		// final boolean isSecure = getConfiguration().getBoolean(
-		// SETTING_KEY_PASSWORD_RESET_LINK_SECURE);
-		// TODO find out how to return just json
-		final String url = "";
-		// final String url = routes.Signup.resetPassword(token).absoluteURL(
-		// ctx.request(), isSecure);
-		
+		final boolean isSecure = getConfiguration().getBoolean(
+		 SETTING_KEY_PASSWORD_RESET_LINK_SECURE);
+		final String url = Play.application().configuration().getString(GlobalData.CONFIG_FORGOT_PASSWORD_URL_BASE)+token;
```

Comment:
We should have a fallback in case the platform is used with a UI that does not provide a form for resetting the password. Let's create a simple one using play views.